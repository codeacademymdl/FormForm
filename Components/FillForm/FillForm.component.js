import React, { Component } from 'react';
import {
  Text, View, TextInput,
} from 'react-native';

import styles from './FillForm.style';


export default class FillForm extends Component {
  render() {
    return (
      <View style={styles.main}>
        <Text style={styles.content}>this is Fill Form</Text>
      </View>

    );
  }
}
