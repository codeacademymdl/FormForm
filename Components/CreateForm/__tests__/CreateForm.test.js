import React from 'react';

import { shallow } from 'enzyme';
import renderer from 'react-test-renderer';
import FillForm from '../CreateForm.component';


describe('FillForm', () => {
  it('renders correctly', () => {
    const tree = renderer
      .create(<FillForm />)
      .toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('checks if state of count changes', () => {
    const wrapper = shallow(<FillForm />);
    expect(wrapper.instance().state.count).toEqual(0);
    wrapper.instance().counterIncrement();
    expect(wrapper.instance().state.count).toEqual(1);
  });

  it('checks if onChangeAddField stores the text in array', () => {
    const wrapper = shallow(<FillForm />);
    expect(wrapper.instance().state.fields).toEqual([]);
    wrapper.instance().onChangeAddField('hello', 0);
    expect(wrapper.instance().state.fields).toEqual(['hello']);
  });

  it('checks if onChangeAddField stores the multiple texts in array', () => {
    const wrapper = shallow(<FillForm />);
    expect(wrapper.instance().state.fields).toEqual([]);
    wrapper.instance().onChangeAddField('hello', 0);
    wrapper.instance().onChangeAddField('bye', 1);
    expect(wrapper.instance().state.fields).toEqual(['hello', 'bye']);
  });

  it('checks if counterIncrement function triggered', () => {
    const wrapper = shallow(<FillForm />);
    wrapper.find('TouchableHighlight').first().simulate('press');
  });

  it('checks if onSubmit function triggered', () => {
    const wrapper = shallow(<FillForm />);
    wrapper.find('TouchableHighlight').at(1).simulate('press');
  });
});
