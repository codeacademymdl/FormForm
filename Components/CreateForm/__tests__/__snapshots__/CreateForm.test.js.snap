// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FillForm renders correctly 1`] = `
<RCTSafeAreaView
  emulateUnlessSupported={true}
>
  <RCTScrollView>
    <View>
      <View
        style={
          Object {
            "alignItems": "center",
            "backgroundColor": "#4154af",
            "flexDirection": "row",
            "height": 60,
            "width": "100%",
          }
        }
      >
        <Text
          style={
            Object {
              "color": "white",
              "fontSize": 20,
              "padding": 10,
            }
          }
        >
          Create Form
        </Text>
      </View>
      <View
        style={
          Object {
            "flex": 1,
            "justifyContent": "center",
          }
        }
      >
        <View
          style={
            Object {
              "borderBottomColor": "#f5f5f6",
              "borderBottomWidth": 2,
              "marginTop": 10,
            }
          }
        >
          <TextInput
            allowFontScaling={true}
            onChangeText={[Function]}
            placeholder="Form Name"
            style={
              Object {
                "height": 40,
              }
            }
            underlineColorAndroid="transparent"
          />
        </View>
        <View
          style={
            Object {
              "flexDirection": "row",
              "justifyContent": "flex-end",
              "margin": 20,
            }
          }
        >
          <View
            accessible={true}
            isTVSelectable={true}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={
              Object {
                "alignItems": "center",
                "backgroundColor": "#4154af",
                "height": 35,
                "padding": 10,
                "width": 100,
              }
            }
          >
            <Text
              style={
                Object {
                  "color": "white",
                }
              }
            >
              Add Field
            </Text>
          </View>
        </View>
        <View
          style={
            Object {
              "alignItems": "center",
              "backgroundColor": "#4154af",
              "flexDirection": "column",
              "height": 50,
              "marginTop": "95%",
              "width": "100%",
            }
          }
        >
          <View
            accessible={true}
            isTVSelectable={true}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={null}
          >
            <Text
              style={
                Object {
                  "color": "white",
                  "fontSize": 20,
                  "padding": 20,
                }
              }
            >
              Save
            </Text>
          </View>
        </View>
      </View>
    </View>
  </RCTScrollView>
</RCTSafeAreaView>
`;
